src/main/java/com/uid2/shared/secure/NitroAttestationProvider.java

```diff
@@ -2,25 +2,33 @@
 
 import com.uid2.shared.secure.nitro.AttestationDocument;
 import com.uid2.shared.secure.nitro.AttestationRequest;
-import com.uid2.shared.secure.IAttestationProvider;
 
+import java.net.MalformedURLException;
+import java.net.URL;
 import java.util.Arrays;
 import java.util.Base64;
 import java.util.Collection;
 import java.util.HashSet;
 import java.util.Set;
 import java.util.stream.Collectors;
 
+import com.uid2.shared.util.UrlEquivalenceValidator;
 import io.vertx.core.AsyncResult;
 import io.vertx.core.Future;
 import io.vertx.core.Handler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public class NitroAttestationProvider implements IAttestationProvider {
```

Comment:
A more general question here, but would it be better to change the name of `IAttestationProvider` to `ISecureAttestationProvider`? We already have an existing `IAttestationProvider` in the uid2-attestation-api repo, and the one that `NitroAttestationProvider` inherits is for enclaves. It got a bit confusing reviewing the change in `NoAttestationProvider` vs `NitroAttestationProvider` with the change to `getAttestationRequest()`

Reply:
I agree, and have renamed it to ICoreAttestationService